### DISCUSSION TITLE
How does github mod mod loader checking work?

### DISCUSSION THREAD

**Discussion by JustSimplyKyle** [gorilla-devs/ferium#58] [Q&A] [answered]
For ~~almost~~ all the github mods I use, I have to add `--dont-check-modloader`. How does the current checking works?

**Comment by theRookieCoder** [comment, marked as answer]
The code is [here](https://github.com/theRookieCoder/libium/blob/main/src/check.rs#L101). It basically checks whether the asset name has the mod loader name in it

**Comment by JustSimplyKyle** [comment]
This doesn't work a lot of times tho, since most build methods will not output modloader in file name.

**Reply by theRookieCoder** [reply]
Yeah true. How else do you think we can check mod loader? Should we check the release description too or something?

**Reply by JustSimplyKyle** [reply]
These tags, but not every mod has this tho, but it can improve the check
https://github.com/topics/fabric-mod
https://github.com/topics/fabricmc-mod
https://github.com/topics/fabricmc
Second one is more complicated, you can know what mod by checking the `gradle.proprities` 
IF it has `fabric_version=a-fabric-version` it is a fabric mod. But for forge? Maybe just default that if a mod's `gradle.proprieties` doesn't contain fabric_version, it means that it is a forge mod. 
And a speical case for iris, since it uses the branchyura build system

**Reply by theRookieCoder** [reply]
We could implement the topics method. Basically use these [fabric topics](https://github.com/search?q=fabric&type=topics) and [forge topics](https://github.com/search?q=forge&type=topics)

**Reply by JustSimplyKyle** [reply]
https://github.com/topics/minecraft-forge-mod
https://github.com/topics/minecraft-forge
https://github.com/topics/forge(Since it's only forge, may contain other non minecraft stuff)
https://github.com/topics/forge-mod
I don't think we should add all the topics in fabric or forge, since that would be a lot of unnecceary checking
